concurrent/src/main/scala/tofu/concurrent/MakeRef.scala

```diff
@@ -3,21 +3,61 @@ package tofu.concurrent
 import cats.effect.Sync
 import cats.effect.concurrent.Ref
 
+/**
+  * Effectful making and initialization for `Ref`.
+  * A 'Ref' instance will be initialized in the `I[_]' effect.
+  * 'F[_]' is the effect in which 'Ref' will work.
+  */
 trait MakeRef[I[_], F[_]] {
   def refOf[A](a: A): I[Ref[F, A]]
 }
 
 object Refs {
+
+  /**
+    * Creates a `MakeRef[F,F]` when both effect constructors are the same.
+    *
+    * @param agents an given instance of `Refs[F]' (type alias for `MakeRef[F, F]`)
+    * @return instace of [[Applier[F, F]]]
+    */
   def apply[F[_]](implicit agents: Refs[F]): MakeRef.Applier[F, F] = new MakeRef.Applier[F, F](agents)
 }
 
 object MakeRef {
+
+  /**
+    * Makes a `Ref` instance
+    *
+    * Uses the
+    * [[https://tpolecat.github.io/2015/07/30/infer.html Kinda-Curried Type Parameters]]
```

Comment:
This blogpost is a bit outdated. There is a description of this technique in cats docs (which mentions Rob's post in turn): https://typelevel.org/cats/guidelines.html#partially-applied-type-params

Reply:
Yes, but, IMHO, this post is more complicated and uses the Applicative [F]. Ok, I will change link.